notes-with-indexeddb/notes.js

```diff
@@ -0,0 +1,169 @@
+/*
+Feature detection
+ */
+// DON'T use 'var indexedDB = ...' if you're not in a function.
+window.indexedDB = window.indexedDB || window.mozIndexedDB || window.webkitIndexedDB || window.msIndexedDB;
+
+// This line should only be needed if it is needed to support the object's constants for older browsers
+window.IDBTransaction = window.IDBTransaction || window.webkitIDBTransaction || window.msIDBTransaction || {
+    READ_WRITE: 'readwrite'
+};
+
+window.IDBKeyRange = window.IDBKeyRange || window.webkitIDBKeyRange || window.msIDBKeyRange;
+
+if (!window.indexedDB) {
+    window.alert("Your browser doesn't support a stable version of IndexedDB.");
+}
+
+/*
+showErrorMessage from event if available
+ */
+function showErrorMessage(event) {
+    if (event && event.target && event.target.errorCode) {
+        alert('Database error: ' + event.target.errorCode);
+        return;
+    }
+
+    alert('IndexedDB request denied');
+}
+
+/*
+addItem to db
+ */
+function addItem(db, data, success) {
+    var store = db.transaction(['notes'], 'readwrite').objectStore('notes'),
```

Comment:
Nit: Please use `let` rather than `var`, and only declare one variable per `let` keyword. Using one `let` or `var` statement for declarations across multiple lines is a common cause of errors when refactoring.

The same goes for the other functions.